celery/result.py

```diff
@@ -823,8 +824,12 @@ class GroupResult(ResultSet):
     #: List/iterator of results in the group
     results = None
 
-    def __init__(self, id=None, results=None, **kwargs):
+    #: Parent Result of the group, if any
+    parent = None
```

Comment:
ResultBase already defines `parent = None`. Wondering if we should get rid of L818/L828.